source/views/streaming/radio.js

```diff
@@ -34,31 +36,61 @@ export default class KSTOView extends React.PureComponent {
     refreshing: boolean,
     paused: boolean,
     streamError: ?Object,
+    uplinkError: boolean,
+    message: string,
     metadata: Object[],
   } = {
     refreshing: false,
     paused: true,
     streamError: null,
+    uplinkError: false,
+    message: '',
     metadata: [],
   }
 
+  // check the stream uplink status
+  fetchUplinkStatus = async () => {
+    try {
+      await promiseTimeout(6000, fetch(kstoStatus))
+      this.setState(() => ({uplinkError: false, message: ''}))
+    } catch (err) {
+      this.setState(() => ({
+        uplinkError: true,
+        message: 'The KSTO stream is down. Sorry!',
+      }))
+    }
+
+    // If the stream is down or we had an error, pause the player
+    if (this.state.uplinkError) {
+      this.setState(() => ({paused: true}))
+    }
+  }
+
   changeControl = () => {
     this.setState(state => ({paused: !state.paused}))
+
+    // If we try to play...
+    if (this.state.paused) {
```

Comment:
This is not guaranteed to have the new state. I almost think that this block should go into the setState callback.

Reply:
I think I know what I'd like this to do; let me take a stab at it in a bit.